algod/Dockerfile

```diff
@@ -0,0 +1,72 @@
+ARG GO_VERSION=1.17.5
+FROM golang:$GO_VERSION-bullseye as builder
+
+ARG CHANNEL=nightly
+ARG URL=
+ARG BRANCH=
+ARG SHA=
+
+# Basic dependencies.
+ENV HOME /node
+ENV DEBIAN_FRONTEND noninteractive
+RUN apt-get update && \
+    apt-get install -y \
+    apt-utils \
+    bsdmainutils \
+    curl \
+    git \
+    git-core \
+    python3
+
+# Copy lots of things into the container. The gitignore indicates which directories.
+COPY . /node/files
+
+# Install algod binaries.
+RUN /node/files/build/install.sh \
+    -p "/node/bin" \
+    -d "/node/data" \
+    -c "${CHANNEL}" \
+    -u "${URL}" \
+    -b "${BRANCH}" \
+    -s "${SHA}"
+
+
+# Copy binaries into a clean image
+# TODO: We don't need most of the binaries.
+#       Should we delete everything except goal/algod/algocfg/tealdbg?
+FROM debian:bullseye-slim as final
```

Comment:
Alpine can be used if we solve this issue:
https://github.com/algorand/go-algorand/issues/3913